src/onegov/core/collection.py

```diff
@@ -7,89 +7,133 @@
 from onegov.core.orm import func
 
 
-class GenericCollection:
+from typing import Any, Generic, Literal, TypeVar, TYPE_CHECKING
+if TYPE_CHECKING:
+    from abc import abstractmethod
+    from collections.abc import Collection, Iterable, Iterator, Sequence
+    from sqlalchemy import Column
+    from sqlalchemy.orm import Query, Session
+    from sqlalchemy.sql.elements import ClauseElement
+    from typing_extensions import Self
+    from uuid import UUID
 
-    def __init__(self, session):
+    from onegov.core.orm import Base
+    from onegov.form.core import Form
```

Comment:
For some reason this introduces a runtime dependency, even though it's only imported when type checking...

Maybe this requires using sys.import?!

Reply:
Unfortunately if we use a dynamic way to import the module, mypy will not understand what type the imported symbol is, so for now we just depend on wtforms.Form, even though we need our own Form class which implements `get_useful_data`